include/cuco/detail/open_addressing/open_addressing_impl.cuh

```diff
@@ -539,6 +540,50 @@ class open_addressing_impl {
     return counter.load_to_host(stream);
   }
 
+  // TODO docs
+  template <typename Container, typename Predicate>
+  void rehash(Container const& container, Predicate const& is_filled, cuda_stream_ref stream)
+  {
+    this->rehash_async(container, is_filled, stream);
+    stream.synchronize();
+  }
+
+  template <typename Container, typename Predicate>
+  void rehash(extent_type extent,
+              Container const& container,
+              Predicate const& is_filled,
+              cuda_stream_ref stream)
+  {
+    this->rehash_async(extent, container, is_filled, stream);
+    stream.synchronize();
+  }
+
+  template <typename Container, typename Predicate>
+  void rehash_async(Container const& container, Predicate const& is_filled, cuda_stream_ref stream)
```

Comment:
I would prefer passing the non-owning container_ref over reference of the container
```suggestion
  void rehash_async(Ref container_ref, Predicate const& is_filled, cuda_stream_ref stream)
```

Reply:
Yeah, I tried that initially. The problem is that the ref we pass to the function uses the old storage ref, not the newly created storage. By passing the container itself, and after swapping in the new storage, I can simply create a new ref from the container with the correct underlying storage.

Reply:
Make sense